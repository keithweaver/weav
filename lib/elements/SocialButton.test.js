/* eslint-disable */
// test file
import React from 'react';
import { shallow } from 'enzyme';


import SocialButton from './SocialButton';

describe('SocialButton', () => {
  test('Shallow render of the SocialButton. Basic render, no props.', () => {
    const wrapper = shallow(
      <SocialButton />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('Shallow render of the SocialButton with custom text.', () => {
    const wrapper = shallow(
      <SocialButton
        text="Foobar"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('Shallow render of the SocialButton with text and text styling.', () => {
    const wrapper = shallow(
      <SocialButton
        text="Foobar"
        textStyle={{
          color: 'red',
        }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('Shallow render of the SocialButton with styling.', () => {
    const wrapper = shallow(
      <SocialButton
        style={{
          borderRadius: 10,
        }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('Shallow render of the SocialButton with icon.', () => {
    const wrapper = shallow(
      <SocialButton
        icon="http://randomimage.com"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('Shallow render of the SocialButton with icon and alt text.', () => {
    const wrapper = shallow(
      <SocialButton
        altText="About the stuff"
        icon="http://anotherimg.com"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('Shallow render of the SocialButton with icon & icon styling.', () => {
    const wrapper = shallow(
      <SocialButton
        icon="http://exampleimg.com"
        iconStyle={{
          height: 20,
        }}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('Shallow render of the SocialButton with background color.', () => {
    const wrapper = shallow(
      <SocialButton
        backgroundColor="#2e2e2e"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('Shallow render of the SocialButton with onClick function.', () => {
    const wrapper = shallow(
      <SocialButton
        onClick={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
