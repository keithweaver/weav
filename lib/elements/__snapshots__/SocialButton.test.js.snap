// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialButton Shallow render of the SocialButton with background color. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SocialButton
    altText=""
    backgroundColor="#2e2e2e"
    icon={null}
    iconStyle={Object {}}
    onClick={[Function]}
    style={Object {}}
    text=""
    textStyle={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <span
          style={
            Object {
              "color": "red",
              "display": "inline-block",
              "fontSize": 14,
              "paddingLeft": 8,
              "verticalAlign": "middle",
            }
          }
        >
          
        </span>,
      ],
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "#2e2e2e",
        "borderRadius": 10,
        "borderWidth": 0,
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
        "textAlign": "left",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "style": Object {
            "color": "red",
            "display": "inline-block",
            "fontSize": 14,
            "paddingLeft": 8,
            "verticalAlign": "middle",
          },
        },
        "ref": null,
        "rendered": "",
        "type": "span",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <span
            style={
              Object {
                "color": "red",
                "display": "inline-block",
                "fontSize": 14,
                "paddingLeft": 8,
                "verticalAlign": "middle",
              }
            }
          >
            
          </span>,
        ],
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "#2e2e2e",
          "borderRadius": 10,
          "borderWidth": 0,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
          "textAlign": "left",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "style": Object {
              "color": "red",
              "display": "inline-block",
              "fontSize": 14,
              "paddingLeft": 8,
              "verticalAlign": "middle",
            },
          },
          "ref": null,
          "rendered": "",
          "type": "span",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SocialButton Shallow render of the SocialButton with custom text. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SocialButton
    altText=""
    backgroundColor=""
    icon={null}
    iconStyle={Object {}}
    onClick={[Function]}
    style={Object {}}
    text="Foobar"
    textStyle={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <span
          style={
            Object {
              "color": "#ffffff",
              "display": "inline-block",
              "fontSize": 14,
              "paddingLeft": 8,
              "verticalAlign": "middle",
            }
          }
        >
          Foobar
        </span>,
      ],
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "",
        "borderRadius": 3,
        "borderWidth": 0,
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
        "textAlign": "left",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Foobar",
          "style": Object {
            "color": "#ffffff",
            "display": "inline-block",
            "fontSize": 14,
            "paddingLeft": 8,
            "verticalAlign": "middle",
          },
        },
        "ref": null,
        "rendered": "Foobar",
        "type": "span",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <span
            style={
              Object {
                "color": "#ffffff",
                "display": "inline-block",
                "fontSize": 14,
                "paddingLeft": 8,
                "verticalAlign": "middle",
              }
            }
          >
            Foobar
          </span>,
        ],
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "",
          "borderRadius": 3,
          "borderWidth": 0,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
          "textAlign": "left",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Foobar",
            "style": Object {
              "color": "#ffffff",
              "display": "inline-block",
              "fontSize": 14,
              "paddingLeft": 8,
              "verticalAlign": "middle",
            },
          },
          "ref": null,
          "rendered": "Foobar",
          "type": "span",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SocialButton Shallow render of the SocialButton with icon & icon styling. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SocialButton
    altText=""
    backgroundColor=""
    icon="http://exampleimg.com"
    iconStyle={
      Object {
        "height": 20,
      }
    }
    onClick={[Function]}
    style={Object {}}
    text=""
    textStyle={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt=""
          src="http://exampleimg.com"
          style={
            Object {
              "display": "inline-block",
              "height": 20,
              "verticalAlign": "middle",
            }
          }
        />,
        <span
          style={
            Object {
              "color": "red",
              "display": "inline-block",
              "fontSize": 14,
              "paddingLeft": 8,
              "verticalAlign": "middle",
            }
          }
        >
          
        </span>,
      ],
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "",
        "borderRadius": 10,
        "borderWidth": 0,
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
        "textAlign": "left",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "src": "http://exampleimg.com",
          "style": Object {
            "display": "inline-block",
            "height": 20,
            "verticalAlign": "middle",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "style": Object {
            "color": "red",
            "display": "inline-block",
            "fontSize": 14,
            "paddingLeft": 8,
            "verticalAlign": "middle",
          },
        },
        "ref": null,
        "rendered": "",
        "type": "span",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            src="http://exampleimg.com"
            style={
              Object {
                "display": "inline-block",
                "height": 20,
                "verticalAlign": "middle",
              }
            }
          />,
          <span
            style={
              Object {
                "color": "red",
                "display": "inline-block",
                "fontSize": 14,
                "paddingLeft": 8,
                "verticalAlign": "middle",
              }
            }
          >
            
          </span>,
        ],
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "",
          "borderRadius": 10,
          "borderWidth": 0,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
          "textAlign": "left",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "http://exampleimg.com",
            "style": Object {
              "display": "inline-block",
              "height": 20,
              "verticalAlign": "middle",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "style": Object {
              "color": "red",
              "display": "inline-block",
              "fontSize": 14,
              "paddingLeft": 8,
              "verticalAlign": "middle",
            },
          },
          "ref": null,
          "rendered": "",
          "type": "span",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SocialButton Shallow render of the SocialButton with icon and alt text. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SocialButton
    altText="About the stuff"
    backgroundColor=""
    icon="http://anotherimg.com"
    iconStyle={Object {}}
    onClick={[Function]}
    style={Object {}}
    text=""
    textStyle={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="About the stuff"
          src="http://anotherimg.com"
          style={
            Object {
              "display": "inline-block",
              "height": 20,
              "verticalAlign": "middle",
            }
          }
        />,
        <span
          style={
            Object {
              "color": "red",
              "display": "inline-block",
              "fontSize": 14,
              "paddingLeft": 8,
              "verticalAlign": "middle",
            }
          }
        >
          
        </span>,
      ],
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "",
        "borderRadius": 10,
        "borderWidth": 0,
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
        "textAlign": "left",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "About the stuff",
          "src": "http://anotherimg.com",
          "style": Object {
            "display": "inline-block",
            "height": 20,
            "verticalAlign": "middle",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "style": Object {
            "color": "red",
            "display": "inline-block",
            "fontSize": 14,
            "paddingLeft": 8,
            "verticalAlign": "middle",
          },
        },
        "ref": null,
        "rendered": "",
        "type": "span",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="About the stuff"
            src="http://anotherimg.com"
            style={
              Object {
                "display": "inline-block",
                "height": 20,
                "verticalAlign": "middle",
              }
            }
          />,
          <span
            style={
              Object {
                "color": "red",
                "display": "inline-block",
                "fontSize": 14,
                "paddingLeft": 8,
                "verticalAlign": "middle",
              }
            }
          >
            
          </span>,
        ],
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "",
          "borderRadius": 10,
          "borderWidth": 0,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
          "textAlign": "left",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "About the stuff",
            "src": "http://anotherimg.com",
            "style": Object {
              "display": "inline-block",
              "height": 20,
              "verticalAlign": "middle",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "style": Object {
              "color": "red",
              "display": "inline-block",
              "fontSize": 14,
              "paddingLeft": 8,
              "verticalAlign": "middle",
            },
          },
          "ref": null,
          "rendered": "",
          "type": "span",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SocialButton Shallow render of the SocialButton with icon. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SocialButton
    altText=""
    backgroundColor=""
    icon="http://randomimage.com"
    iconStyle={Object {}}
    onClick={[Function]}
    style={Object {}}
    text=""
    textStyle={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt=""
          src="http://randomimage.com"
          style={
            Object {
              "display": "inline-block",
              "height": 20,
              "verticalAlign": "middle",
            }
          }
        />,
        <span
          style={
            Object {
              "color": "red",
              "display": "inline-block",
              "fontSize": 14,
              "paddingLeft": 8,
              "verticalAlign": "middle",
            }
          }
        >
          
        </span>,
      ],
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "",
        "borderRadius": 10,
        "borderWidth": 0,
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
        "textAlign": "left",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "src": "http://randomimage.com",
          "style": Object {
            "display": "inline-block",
            "height": 20,
            "verticalAlign": "middle",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "style": Object {
            "color": "red",
            "display": "inline-block",
            "fontSize": 14,
            "paddingLeft": 8,
            "verticalAlign": "middle",
          },
        },
        "ref": null,
        "rendered": "",
        "type": "span",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            src="http://randomimage.com"
            style={
              Object {
                "display": "inline-block",
                "height": 20,
                "verticalAlign": "middle",
              }
            }
          />,
          <span
            style={
              Object {
                "color": "red",
                "display": "inline-block",
                "fontSize": 14,
                "paddingLeft": 8,
                "verticalAlign": "middle",
              }
            }
          >
            
          </span>,
        ],
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "",
          "borderRadius": 10,
          "borderWidth": 0,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
          "textAlign": "left",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "http://randomimage.com",
            "style": Object {
              "display": "inline-block",
              "height": 20,
              "verticalAlign": "middle",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "style": Object {
              "color": "red",
              "display": "inline-block",
              "fontSize": 14,
              "paddingLeft": 8,
              "verticalAlign": "middle",
            },
          },
          "ref": null,
          "rendered": "",
          "type": "span",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SocialButton Shallow render of the SocialButton with onClick function. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SocialButton
    altText=""
    backgroundColor=""
    icon={null}
    iconStyle={Object {}}
    onClick={[Function]}
    style={Object {}}
    text=""
    textStyle={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <span
          style={
            Object {
              "color": "red",
              "display": "inline-block",
              "fontSize": 14,
              "paddingLeft": 8,
              "verticalAlign": "middle",
            }
          }
        >
          
        </span>,
      ],
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "",
        "borderRadius": 10,
        "borderWidth": 0,
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
        "textAlign": "left",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "style": Object {
            "color": "red",
            "display": "inline-block",
            "fontSize": 14,
            "paddingLeft": 8,
            "verticalAlign": "middle",
          },
        },
        "ref": null,
        "rendered": "",
        "type": "span",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <span
            style={
              Object {
                "color": "red",
                "display": "inline-block",
                "fontSize": 14,
                "paddingLeft": 8,
                "verticalAlign": "middle",
              }
            }
          >
            
          </span>,
        ],
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "",
          "borderRadius": 10,
          "borderWidth": 0,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
          "textAlign": "left",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "style": Object {
              "color": "red",
              "display": "inline-block",
              "fontSize": 14,
              "paddingLeft": 8,
              "verticalAlign": "middle",
            },
          },
          "ref": null,
          "rendered": "",
          "type": "span",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SocialButton Shallow render of the SocialButton with styling. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SocialButton
    altText=""
    backgroundColor=""
    icon={null}
    iconStyle={Object {}}
    onClick={[Function]}
    style={
      Object {
        "borderRadius": 10,
      }
    }
    text=""
    textStyle={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <span
          style={
            Object {
              "color": "red",
              "display": "inline-block",
              "fontSize": 14,
              "paddingLeft": 8,
              "verticalAlign": "middle",
            }
          }
        >
          
        </span>,
      ],
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "",
        "borderRadius": 10,
        "borderWidth": 0,
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
        "textAlign": "left",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "style": Object {
            "color": "red",
            "display": "inline-block",
            "fontSize": 14,
            "paddingLeft": 8,
            "verticalAlign": "middle",
          },
        },
        "ref": null,
        "rendered": "",
        "type": "span",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <span
            style={
              Object {
                "color": "red",
                "display": "inline-block",
                "fontSize": 14,
                "paddingLeft": 8,
                "verticalAlign": "middle",
              }
            }
          >
            
          </span>,
        ],
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "",
          "borderRadius": 10,
          "borderWidth": 0,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
          "textAlign": "left",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "style": Object {
              "color": "red",
              "display": "inline-block",
              "fontSize": 14,
              "paddingLeft": 8,
              "verticalAlign": "middle",
            },
          },
          "ref": null,
          "rendered": "",
          "type": "span",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SocialButton Shallow render of the SocialButton with text and text styling. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SocialButton
    altText=""
    backgroundColor=""
    icon={null}
    iconStyle={Object {}}
    onClick={[Function]}
    style={Object {}}
    text="Foobar"
    textStyle={
      Object {
        "color": "red",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <span
          style={
            Object {
              "color": "red",
              "display": "inline-block",
              "fontSize": 14,
              "paddingLeft": 8,
              "verticalAlign": "middle",
            }
          }
        >
          Foobar
        </span>,
      ],
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "",
        "borderRadius": 3,
        "borderWidth": 0,
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
        "textAlign": "left",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Foobar",
          "style": Object {
            "color": "red",
            "display": "inline-block",
            "fontSize": 14,
            "paddingLeft": 8,
            "verticalAlign": "middle",
          },
        },
        "ref": null,
        "rendered": "Foobar",
        "type": "span",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <span
            style={
              Object {
                "color": "red",
                "display": "inline-block",
                "fontSize": 14,
                "paddingLeft": 8,
                "verticalAlign": "middle",
              }
            }
          >
            Foobar
          </span>,
        ],
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "",
          "borderRadius": 3,
          "borderWidth": 0,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
          "textAlign": "left",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Foobar",
            "style": Object {
              "color": "red",
              "display": "inline-block",
              "fontSize": 14,
              "paddingLeft": 8,
              "verticalAlign": "middle",
            },
          },
          "ref": null,
          "rendered": "Foobar",
          "type": "span",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SocialButton Shallow render of the SocialButton. Basic render, no props. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SocialButton
    altText=""
    backgroundColor=""
    icon={null}
    iconStyle={Object {}}
    onClick={[Function]}
    style={Object {}}
    text=""
    textStyle={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <span
          style={
            Object {
              "color": "#ffffff",
              "display": "inline-block",
              "fontSize": 14,
              "paddingLeft": 8,
              "verticalAlign": "middle",
            }
          }
        >
          
        </span>,
      ],
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "",
        "borderRadius": 3,
        "borderWidth": 0,
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
        "textAlign": "left",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "style": Object {
            "color": "#ffffff",
            "display": "inline-block",
            "fontSize": 14,
            "paddingLeft": 8,
            "verticalAlign": "middle",
          },
        },
        "ref": null,
        "rendered": "",
        "type": "span",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <span
            style={
              Object {
                "color": "#ffffff",
                "display": "inline-block",
                "fontSize": 14,
                "paddingLeft": 8,
                "verticalAlign": "middle",
              }
            }
          >
            
          </span>,
        ],
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "",
          "borderRadius": 3,
          "borderWidth": 0,
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
          "textAlign": "left",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "style": Object {
              "color": "#ffffff",
              "display": "inline-block",
              "fontSize": 14,
              "paddingLeft": 8,
              "verticalAlign": "middle",
            },
          },
          "ref": null,
          "rendered": "",
          "type": "span",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
